File: python/a06.py
N, Q = map(int, input().split())
A = list(map(int, input().split()))
A = [0] + A
S = [0]
for a in A[1:]:
    S.append(S[-1] + a)


for _ in range(Q):
    L, R = map(int, input().split())
    print(S[R] - S[L-1])

File: rust/a06.rs
use proconio::input;
fn main() {
    input! {
        n: usize,
        q: usize,
        a: [usize; n],
        lr: [(usize,usize); q],
    }
    let mut s = vec![0];
    for i in 0..n {
        s.push(s[i] + a[i]);
    }
    for (l, r) in lr.iter() {
        println!("{}", s[*r] - s[*l-1]);
    }
}
